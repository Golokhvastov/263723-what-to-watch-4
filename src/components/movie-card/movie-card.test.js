import React from "react";
import renderer from "react-test-renderer";
import MovieCard from "./movie-card.jsx";

const movie = {
  title: `Тест1`,
  previewImage: `test1.jpg`,
  preview: `test2.jpg`,
};

describe(`MovieCard render correctly`, () => {
  it(`isPlaying = true`, () => {
    const tree = renderer
      .create(
          <MovieCard
            movie = {movie}
            onTitleClick = {() => {}}
            onMouseEnter = {() => {}}
            onMouseLeave = {() => {}}
            onCardMouseEnter = {() => {}}
            onCardMouseLeave = {() => {}}
            renderVideo = {() => {}}
            isPlaying = {true}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`isPlaying = false`, () => {
    const tree = renderer
      .create(
          <MovieCard
            movie = {movie}
            onTitleClick = {() => {}}
            onMouseEnter = {() => {}}
            onMouseLeave = {() => {}}
            onCardMouseEnter = {() => {}}
            onCardMouseLeave = {() => {}}
            renderVideo = {() => {}}
            isPlaying = {false}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
