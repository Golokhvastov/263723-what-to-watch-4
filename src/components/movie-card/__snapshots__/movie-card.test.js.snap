// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard render correctly isPlaying = false 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Тест1"
      height="175"
      src="test1.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Тест1
    </a>
  </h3>
</article>
`;

exports[`MovieCard render correctly isPlaying = true 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  />
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Тест1
    </a>
  </h3>
</article>
`;
